src/main/java/com/pinterest/secor/uploader/Uploader.java

```diff
@@ -181,12 +182,31 @@ protected void trimFiles(TopicPartition topicPartition, long startOffset) throws
         }
     }
 
+    /***
+     * If the topic is in the list of topics to upload at a specific time. For example at a minute mark.
+     * @param topicPartition
+     * @return
+     * @throws Exception
+     */
+    private boolean isRequiredToUploadAtTime(TopicPartition topicPartition) throws Exception{
+        final String topic = topicPartition.getTopic();
+        final String topicFilter = mConfig.getKafkaTopicUploadAtMinuteMarkFilter();
+        if (topicFilter == null || topicFilter.isEmpty()){ return false; }
+        if (topic.matches(topicFilter)){
+            if (DateTime.now().minuteOfHour().get() == mConfig.getUploadMinuteMark()){
+               return true;
+            }
+        }
+        return false;
+    }
+
     private void checkTopicPartition(TopicPartition topicPartition) throws Exception {
         final long size = mFileRegistry.getSize(topicPartition);
         final long modificationAgeSec = mFileRegistry.getModificationAgeSec(topicPartition);
         LOG.debug("size: " + size + " modificationAge: " + modificationAgeSec);
         if (size >= mConfig.getMaxFileSizeBytes() ||
-                modificationAgeSec >= mConfig.getMaxFileAgeSeconds()) {
+                modificationAgeSec >= mConfig.getMaxFileAgeSeconds() ||
```

Comment:
Is this because this is a slow volume topic?  Otherwise modificationAgeSec would trigger the upload too.